pkg/chunked/dump/dump.go

```diff
@@ -0,0 +1,219 @@
+package dump
+
+import (
+	"fmt"
+	"io"
+	"strings"
+	"time"
+	"unicode"
+
+	"github.com/containers/storage/pkg/chunked/internal"
+	"golang.org/x/sys/unix"
+)
+
+const (
+	NOESCAPE_SPACE = 1 << iota
+	ESCAPE_EQUAL
+	ESCAPE_LONE_DASH
+	ESCAPE_STANDARD
+)
+
+func escaped(val string, length int, escape int) string {
```

Comment:
I understand this is a direct port of the C code, but you can really drop the length here, as in golang the string comes with its own length (and you can easily create substrings that also come with their own length).